internal/types/command.go

```diff
@@ -67,7 +67,22 @@ type CommandData struct {
 	 * Arguments that the command handler should be
 	 * invoked with.
 	 */
-	Arguments []any
+	Arguments     []any
+	GroupingKey   Key
```

Comment:
how are the Grouping properties related to CommandData? 
Aren't they only necessary for codeaction? or are there other use cases where we need them ?